.github/workflows/build.yml

```diff
@@ -113,6 +113,59 @@ jobs:
           which cargo-cache || cargo install cargo-cache
           cargo cache trim -l 1G
 
+  s3_test:
+    runs-on: ubuntu-latest
+    env:
+      RUSTFLAGS: "-C target-cpu=skylake"
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/cache@v2
+      with:
+        path: |
+          ~/.cargo/bin/
+          ~/.cargo/registry/index/
+          ~/.cargo/registry/cache/
+          ~/.cargo/git/db/
+          target/
+        key: s3-1-${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock', '**/Cargo.toml') }}
+        restore-keys: |
+          s3-1-${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock', '**/Cargo.toml') }}
+          s3-1-${{ runner.os }}-cargo-
+    - name: Install nightly rust
+      uses: actions-rs/toolchain@v1
+      with:
+        profile: default
+        # toolchain: nightly
+        toolchain: nightly-${{ env.RUST_TC_NIGHTLY_VER }}
+        override: true
+    - name: Install minio server
+      run: |
+        docker run -d -p 9000:9000 quay.io/minio/minio server /data
+    - name: Install minio client
+      run: |
+        curl https://dl.min.io/client/mc/release/linux-amd64/mc --create-dirs -o $HOME/minio-binaries/mc
+        chmod +x $HOME/minio-binaries/mc
+    - name: Create bucket and copy test data
+      run: |
+        $HOME/minio-binaries/mc alias set local http://127.0.0.1:9000 minioadmin minioadmin
+        $HOME/minio-binaries/mc mb local/test-data
+        $HOME/minio-binaries/mc cp test_data/blogs.parquet local/test-data
+    - name: Start roapi and Query
+      env:
+        AWS_DEFAULT_REGION: "us-east-1"
+        AWS_ACCESS_KEY_ID: minioadmin
+        AWS_SECRET_ACCESS_KEY: minioadmin
+        AWS_ENDPOINT_URL: http://127.0.0.1:9000
+      uses: BerniWittmann/background-server-action@v1.0.4
```

Comment:
there are several tries to run `roapi` in background, however, it is quite hard in Github Actions, until I found this `background-server-action`

Reply:
interesting, shouldn't we be able to simply run roapi as a background job in a bash script (e.g. query_blogs.sh)?

Reply:
it is a bit annoying and hard to debug, probably because I am not that familiar with Github Actions. (e.g. I didn't find pause and ssh that were available in CircleCI or Buildkite)

I try run it as background in commit a552fa954013f0c81bf17d8c09ff77c5b9986bdf, but looks like the process got exit immediately (while my local terminal works)